src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs

```diff
@@ -880,6 +902,164 @@ public void TestMergeChanges_NoMiddleMan()
         }
 
+        [Fact]
+        [WorkItem(47234, "https://github.com/dotnet/roslyn/issues/47234")]
+        public void DebuggerDisplay()
+        {
+            Assert.Equal("new TextChange(new TextSpan(0, 0), null)", default(TextChange).GetDebuggerDisplay());
+            Assert.Equal("new TextChange(new TextSpan(0, 1), \"abc\")", new TextChange(new TextSpan(0, 1), "abc").GetDebuggerDisplay());
+            Assert.Equal("new TextChange(new TextSpan(0, 1), (NewLength = 10))", new TextChange(new TextSpan(0, 1), "0123456789").GetDebuggerDisplay());
+        }
+
+        [Fact]
+        [WorkItem(47234, "https://github.com/dotnet/roslyn/issues/47234")]
+        public void Fuzz()
+        {
+            var random = new Random();
+
+            // Adjust upper bound as needed to generate a simpler reproducer for an error scenario
+            var originalText = SourceText.From(string.Join("", Enumerable.Range(0, random.Next(10))));
+
+            foreach (var iteration in Enumerable.Range(0, 100000))
+            {
+                var editedLength = originalText.Length;
+                ArrayBuilder<TextChange> oldChangesBuilder = ArrayBuilder<TextChange>.GetInstance();
+
+                const int maxEditLength = 3;
+                const int maxSkipLength = 2;
+                // generate sequence of "old edits" which meet invariants
+                for (int i = 0; i < originalText.Length; i += 1 + random.Next(maxSkipLength))
+                {
+                    var newText = string.Join("", Enumerable.Repeat('a', random.Next(maxEditLength)));
+                    var newChange = new TextChange(new TextSpan(i, length: random.Next(Math.Min(originalText.Length - i, maxEditLength))), newText);
+                    i = newChange.Span.End;
+
+                    editedLength = editedLength - newChange.Span.Length + newChange.NewText.Length;
+                    oldChangesBuilder.Add(newChange);
+
+                    // Adjust as needed to generate a simpler reproducer for an error scenario
+                    if (oldChangesBuilder.Count == 5) break;
+                }
+
+                var change1 = originalText.WithChanges(oldChangesBuilder);
+
+                ArrayBuilder<TextChange> newChangesBuilder = ArrayBuilder<TextChange>.GetInstance();
+
+                // generate sequence of "new edits" which meet invariants
+                for (int i = 0; i < editedLength; i += 1 + random.Next(maxSkipLength))
+                {
+                    var newText = string.Join("", Enumerable.Repeat('b', random.Next(maxEditLength)));
+                    var newChange = new TextChange(new TextSpan(i, length: random.Next(Math.Min(editedLength - i, maxEditLength))), newText);
+                    i = newChange.Span.End;
+
+                    newChangesBuilder.Add(newChange);
+
+                    // Adjust as needed to generate a simpler reproducer for an error scenario
+                    if (newChangesBuilder.Count == 5) break;
```

Comment:
Is it worth putting 5 into a const since we use it twice?

Reply:
I think each place should be its own "knob" to adjust so to speak. When I was coming up with minimal test cases to repro problems I was seeing it ended up being useful to have 2 old changes and 1 new change, for example.